quarkus/config-api/src/main/java/org/keycloak/config/database/Database.java

```diff
@@ -136,29 +157,41 @@ private String addH2NonKeywords(String jdbcUrl) {
                 "dev-mem", "dev-file"
         ),
         MYSQL("mysql",
+                Enabled.LEGACY_ONLY,
                 "com.mysql.cj.jdbc.MysqlXADataSource",
                 "com.mysql.cj.jdbc.Driver",
                 "org.hibernate.dialect.MySQL8Dialect",
                 "jdbc:mysql://${kc.db-url-host:localhost}:${kc.db-url-port:3306}/${kc.db-url-database:keycloak}${kc.db-url-properties:}",
                 asList("org.keycloak.connections.jpa.updater.liquibase.UpdatedMySqlDatabase")
         ),
         MARIADB("mariadb",
+                Enabled.LEGACY_ONLY,
                 "org.mariadb.jdbc.MariaDbDataSource",
                 "org.mariadb.jdbc.Driver",
                 "org.hibernate.dialect.MariaDBDialect",
                 "jdbc:mariadb://${kc.db-url-host:localhost}:${kc.db-url-port:3306}/${kc.db-url-database:keycloak}${kc.db-url-properties:}",
                 asList("org.keycloak.connections.jpa.updater.liquibase.UpdatedMariaDBDatabase")
         ),
         POSTGRES("postgresql",
+                Enabled.ENABLED,
                 "org.postgresql.xa.PGXADataSource",
                 "org.postgresql.Driver",
                 "io.quarkus.hibernate.orm.runtime.dialect.QuarkusPostgreSQL10Dialect",
                 "jdbc:postgresql://${kc.db-url-host:localhost}:${kc.db-url-port:5432}/${kc.db-url-database:keycloak}${kc.db-url-properties:}",
-                asList("liquibase.database.core.PostgresDatabase", "liquibase.database.core.CockroachDatabase",
-                        "org.keycloak.connections.jpa.updater.liquibase.PostgresPlusDatabase"),
+                asList("liquibase.database.core.PostgresDatabase", "org.keycloak.connections.jpa.updater.liquibase.PostgresPlusDatabase"),
                 "postgres"
         ),
+        COCKROACH(POSTGRES.databaseKind, //needs to be aligned with https://quarkus.io/guides/datasource#default-datasource
+                Enabled.MAP_STORE_ONLY,
+                POSTGRES.xaDriver,
+                POSTGRES.nonXaDriver,
+                "org.hibernate.dialect.CockroachDialect",
```

Comment:
AFAIK this should be a valid Hibernate class name. As there is no `CockroachDialect`, I'm suggesting as the most recent dialect supported in Hibernate 5.x.

At the same time, this is in the Quarkus 2.x branch not used, as we'll override it for the new store with the `JsonbPostgreSQL95Dialect`, so this explains why the tests are green.

```suggestion
                "org.hibernate.dialect.CockroachDB201Dialect",
```

Reply:
Thanks, for pointing that out. You're right, the `CockroachDialect` is supported since Hibernate 6. I'll make sure it's reflected in https://github.com/keycloak/keycloak/pull/16549